net/alertmanager/Portfile

```diff
@@ -0,0 +1,132 @@
+github.tarball_from archive
+
+description         The Alertmanager handles alerts sent by client \
+                    applications such as the Prometheus server.
+
+long_description    AlertManager takes care of deduplicating, grouping, and \
+                    routing them to the correct receiver integration such as \
+                    email, PagerDuty, or OpsGenie. It also takes care of \
+                    silencing and inhibition of alerts.
+
+homepage            https://prometheus.io/docs/alerting/alertmanager/
+
+platforms           darwin
+categories          net
+license             Apache-2
+
+maintainers         {gmail.com:herbygillot @herbygillot} openmaintainer
+
+depends_build       port:curl \
+                    port:go
+
+build.env           GOPATH=${workpath} \
+                    PATH=$env(PATH):${workpath}/bin
+
+build.target        build
+
+use_configure       no
+installs_libs       no
+use_parallel_build  no
+
+checksums \
+  rmd160    626e2f17560ef2ca70edb9fe57b0fc8506936c09 \
+  sha256    8254591e058338b31a023f0eb3ab69de9d1547e0a446d28cab4664609793e4a2 \
+  size      5118311
+
+set svc_name        prometheus-alertmanager
+set prom_user       prometheus
+set am_config_dir   ${prefix}/etc/${svc_name}
+set am_config_file  ${am_config_dir}/config.yml
+set am_data_dir     ${prefix}/var/db/${svc_name}
+set am_share_dir    ${prefix}/share/${svc_name}
+set am_log_dir      ${prefix}/var/log/${svc_name}
+set am_log_file     ${am_log_dir}/${name}.log
+
+add_users           ${prom_user} \
+                    group=${prom_user} \
+                    realname=Prometheus
+
+post-extract {
+    copy  ${filespath}/org.macports.${name}.plist \
+          ${workpath}/org.macports.${name}.plist
+
+    reinplace "s|@NAME@|${name}|g" \
+        ${workpath}/org.macports.${name}.plist
+
+    reinplace "s|@USER@|${prom_user}|g" \
+        ${workpath}/org.macports.${name}.plist
+
+    reinplace "s|@GROUP@|${prom_user}|g" \
+        ${workpath}/org.macports.${name}.plist
+
+    reinplace "s|@PREFIX@|${prefix}|g" \
+        ${workpath}/org.macports.${name}.plist
+
+    reinplace "s|@LOGFILE@|${am_log_file}|g" \
+        ${workpath}/org.macports.${name}.plist
+
+    reinplace "s|@DATADIR@|${am_data_dir}|g" \
+        ${workpath}/org.macports.${name}.plist
+
+    reinplace "s|@CONFIGFILE@|${am_config_file}|g" \
+        ${workpath}/org.macports.${name}.plist
+}
+
+destroot {
+    xinstall -m 755 ${worksrcpath}/${name} ${destroot}${prefix}/bin/${name}
+    xinstall -m 755 ${worksrcpath}/${name} ${destroot}${prefix}/bin/amtool
+
+    xinstall -d 755 ${destroot}${am_config_dir}
+    xinstall -d 755 ${destroot}${am_share_dir}
+    xinstall -d 755 -o ${prom_user} -g ${prom_user} ${destroot}${am_log_dir}
+    xinstall -d 755 -o ${prom_user} -g ${prom_user} ${destroot}${am_data_dir}
+
+    touch ${destroot}${am_log_file}
+    touch ${destroot}${am_config_dir}/.keep
+    touch ${destroot}${am_data_dir}/.keep
```

Comment:
if the purpose of these three lines is only to make sure the directories are not removed in the ```destroot``` phase as it appears to me, then you'd probably want to use ```destrooot.keepdirs``` instead (see the [guide](https://guide.macports.org/#reference.phases.destroot))

Reply:
Thank you @reneeotten, I forgot about `destroot.keepdirs`.  Fixed.